ci/checkLinks.ts

```diff
@@ -1,185 +1,211 @@
+		if (line.startsWith("```")) {
+			multilineCode = !multilineCode;
+			if (line.length > 3 && line.endsWith("```")) multilineCode = !multilineCode;
+		}
+		if (multilineCode) return;
+		const matches = line.matchAll(regex);
+
+		for (const match of matches) {
+			const split = match[index].split("#");
+			let url = split[0].endsWith("/") ? split[0].slice(0, -1) : split[0];
+			if (match[index].startsWith("#")) url = name;
+			if (!valid.has(url)) {
+				results.push({
+					title: `Base url ${chalk.blueBright(url)} does not exist`,
+					startLine: lineNum + 1,
+					startColumn: match.index,
+					endColumn: (match.index ?? 0) + match[0].length,
+				});
+				continue;
+			}
+
+			if (!split[1]) continue;
+			const validAnchors = valid.get(url)!;
+			if (!validAnchors.includes(split[1])) {
+				results.push({
+					title: `Anchor ${chalk.cyan(split[1])} does not exist on ${chalk.blueBright(url)}`,
+					startLine: lineNum + 1,
+					startColumn: match.index,
+					endColumn: (match.index ?? 0) + match[0].length,
+				});
+			}
+		}
+	});
 }
 
-const htmlFiles = importDirectory(`${cwd}/.next/server/pages/`, '.html');
+const htmlFiles = importDirectory(`${cwd}/.next/server/pages/`, ".html");
 
 if (!htmlFiles) {
-  console.error('No links found, ensure that build has been run!');
-  process.exit(1);
+	console.error("No links found, ensure that build has been run!");
+	process.exit(1);
 }
 
 const validLinks = new Map<string, string[]>();
 
-let extLength = '.html'.length;
+let extLength = ".html".length;
 
 for (const [name, raw] of htmlFiles) {
-  const keyName = name.slice(0, -extLength);
-  if (!validLinks.has(keyName)) {
-    validLinks.set(keyName, []);
-  }
-  const validAnchors = validLinks.get(keyName)!;
-  const fragment = JSDOM.fragment(raw);
-  // @ts-ignore
-  const main = fragment.querySelector('main');
-  if (!main) continue;
-  const allIds = main.querySelectorAll('*[id]');
-  for (const node of allIds.values()) {
-    validAnchors.push(node.id);
-  }
+	const keyName = name.slice(0, -extLength);
+	if (!validLinks.has(keyName)) {
+		validLinks.set(keyName, []);
+	}
+	const validAnchors = validLinks.get(keyName)!;
+	const fragment = JSDOM.fragment(raw);
+	// @ts-ignore
+	const main = fragment.querySelector("main");
+	if (!main) continue;
+	const allIds = main.querySelectorAll("*[id]");
+	for (const node of allIds.values()) {
+		validAnchors.push(node.id);
+	}
 }
 
 const results = new Map<string, github.AnnotationProperties[]>();
 
 try {
-  const navFile = '/components/Navigation.tsx'
-  const nav = readFileSync(`${cwd}${navFile}`, 'utf8');
-  const file = nav.split('\n')
-  if (!results.has(navFile)) {
-    results.set(navFile, []);
-  }
-  const ownResults = results.get(navFile)!;
-  scanFile(/(?<!!)href=(["'])(?!(?:https?)|(?:mailto))(.+?)\1/g, 2, navFile.slice(0, -'.tsx'.length), file, validLinks, ownResults);
+	const navFile = "/components/Navigation.tsx";
+	const nav = readFileSync(`${cwd}${navFile}`, "utf8");
```

Comment:
Ok, realistically you should be using path.join for path manipulation 😅

Reply:
oh, forgot to update those when I added the path import